### PR TITLE
Log the best epoch when restoring the best weights in early stopping

### PR DESCRIPTION
**Pull request by harupy**
Log the best epoch when restoring the best weights in early stopping.

### CODE REVIEW

File: keras/callbacks.py [hunk 1]
```diff
@@ -1822,7 +1824,8 @@ def on_epoch_end(self, epoch, logs=None):
       self.model.stop_training = True
       if self.restore_best_weights and self.best_weights is not None:
         if self.verbose > 0:
-          print('Restoring model weights from the end of the best epoch.')
+          print('Restoring model weights from the end of the best epoch (%s).'
```
Comment (qlzh727) [inline keras/callbacks.py:1827]: nit, you can use formatted string like:

print(f'Restoring model weights from the end of the best epoch: {self.best_epoch + 1}')

**Review by qlzh727 — APPROVED** [review]
Thanks for the change.

@rchao, just curious for the usage of print here, it will probably get buried in the log stream if it is not using absl logging.

**Comment by rchao** [comment]
> Thanks for the change.
> 
> @rchao, just curious for the usage of print here, it will probably get buried in the log stream if it is not using absl logging.

Yeah, I would prefer using absl logging so we have consistent timestamps, but since this class is using print elsewhere I'd be fine with using print for now and having a larger change later.

**Review by rchao — APPROVED** [review]
Thanks for the PR!

**Comment by harupy** [comment]
@qlzh727 @rchao Thanks for the review and merge!. Should I create a follow-up RP to address @rchao's comment: https://github.com/keras-team/keras/pull/15197#discussion_r692303599?

**Pull request merged**